src/Stratis.Bitcoin.Features.PoA/PoAMiner.cs

```diff
@@ -1,16 +1,121 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Interfaces;
+using Stratis.Bitcoin.Mining;
+using Stratis.Bitcoin.Utilities;
 
 namespace Stratis.Bitcoin.Features.PoA
 {
-    public class PoAMiner
+    public class PoAMiner : IDisposable
     {
-        // TODO POA. First add creation of block template and later apply signatures and other stuff. We need bare minimum to test syncing and payloads
+        // TODO POA. Implement miner properly later. Right now we need bare minimum to test syncing and payloads.
 
-        public void StartMining()
+        private const bool MineDuringIBD = true; // TODO POA just for tests
```

Comment:
this is bound to be very confusing. Maybe we should never mine during IBD and for the test, we make sure we're not in IBD.

Reply:
It's for testing (but not in unit tests- testing during the development). Without it every time I want to run the miner I'll need to change network so genesis block have timestamp that is close to my machine's timestamp, otherwise it wont mine because we will be in ibd.

Reply:
ok

Reply:
The fix is probably not to create such a flag `MineDuringIBD` but rather in the calculation of difficulty you check the testnet flag on the network class and allow blocks when in testnet.

Also we solved the time issue in POS by injecting a fake datetime provider that will simulate the date for what the tests needs to really no need for this flag.

Reply:
it's not for unit tests, it's for running the node, so injection is just a more difficult approach.
this bool will be removed as soon as the network can run stable.

Reply:
OK but integration tests should be created with injection of override tool and not this flag (if the flag is intended to be deleted that fine with me)